demos/nprk/demo_burgers_nprk.py.rst

```diff
@@ -0,0 +1,106 @@
+Solving the Viscous Burgers' Equation with a Non-Linearly Partitioned RK Scheme
+===============================================================================
+
+Consider the viscous Burgers' equation
+
+.. math::
+
+   \partial_t u + u\, \partial_x u - \epsilon\, \partial_x^2 u = 0
+
+To demonstrate the non-linearly partitioned RK (NPRK) method, various partitionings are listed and compared in:
+
+Buvoli, Southworth "A New Class of Runge-Kutta Methods for Nonlinearly Partitioned Systems", arXiv 2024
+
+The non-linear partitioning is explained here and implemented in Irksome.
+
+A non-linear partitioning equates the time derivative with an operator taking two arguments, :math:`u_t=F(u,u)`.
+Each stage of the NPRK scheme treats one argument implicitly and the other explicitly.
+There are many choices of non-linear partitioning, but
+
+.. math::
+
+   F(u_1, u_2) = - u_2\, \partial_x u_1 + \epsilon\, \partial_x^2 u_1
+
+is selected.
+
+With homogenous Dirichlet boundary conditions, this equation gives rise to the weak form
+
+.. math::
+
+   (v, \partial_t u) + (v, u_2\, \partial_x u_1) + (\partial_x v, \epsilon\, \partial_x u_2) = 0
+
+Following the paper, this equation is solved numerically for :math:`\epsilon=1/200`, :math:`x\in[-2,2]`, :math:`t\in[0,0.6]`
+with :math:`1000` grid points and the initial condition
+
+.. math::
+
+   u(x,0)=e^{-3x^2}
+
+The timestep :math:`\Delta t=6\cdot 10^{-3}` is selected as the middle of the range tested in the referenced paper.
+
+To implement this, begin by importing firedrake::
+
+   from firedrake import *
+
+Additionally, a few basic items and the NPRK time stepper are imported from irksome::
+
+   from irksome import Dt, MeshConstant, NPRKMethod
```

Comment:
`NPRKMethod` doesn't match our naming for other methods.  It would be more helpful to the readier to be chattier, like `NonlinearlyPartitionedRKTimeStepper`, and more parallel to the other schemes we have.